### DISCUSSION TITLE
Composite image with rotation

### DISCUSSION THREAD

**Discussion by CommanderAlchemy** [dlemstra/Magick.NET#1204] [Help]
Hi, I am trying currently to add image onto another with x and y coordinates and rotation. 

w/ followng code:
```c#
dropImage.Rotate(45);
baseImage.Composite(dropImage, 
new PointD(drop.GetXCoordinate(), drop.GetYCoordinate()), CompositeOperator.SrcOver);
```

This will cause a bounding box around the image, causing my original `x` and `y` coordinates being off from my original image. 
How can I rotate and also keep the coordinates so that I can place the image correctly on top the background image. 

I've seen examples of using args, but cannot seem to make it do anything.
```c#
baseImage.Composite(dropImage, 
new PointD(drop.GetXCoordinate(), drop.GetYCoordinate()), CompositeOperator.SrcOver, $"-rotate {45}");
```

![image](https://user-images.githubusercontent.com/3089330/174618578-dad46414-8a3c-483c-818a-880d3579101e.png)

**Comment by dlemstra** [comment]
Can you share the two images?

**Reply by CommanderAlchemy** [reply]
> Can you share the two images?

Sure they are just my test images.

DropImage
![55c5f4b0-f1d2-4f5c-b8d0-6cbc4246c4a0](https://user-images.githubusercontent.com/3089330/174738471-39e9cb09-3203-4e28-bf7a-56b58c47187b.png)

BaseImage
![63e5c177-248a-4c70-9aae-923fa75c7020](https://user-images.githubusercontent.com/3089330/174738428-20755bf9-84bd-4822-95f7-d76eaed74773.png)

The previous image was just a compilation on 3 different rotations. The white background is occurring when using odd degrees but is fixed when using `image.BackgroundColor = MagickColors.Transparent;` Just kept it to see how the rotation works more easily. 

**TLDR** the second arrow (right one) in the pictures, I want the image to rotate around its coordinates and to be placed where the first arrow is (left one) is.  (thats the GetY/XCoordinates) I got stored.

Right now the only way I can think of is to do some trigonometry to calculate x and y based on the angle baseWidth/Height and Width/Height + rotation but it feels a bit overkill.

**Reply by CommanderAlchemy** [reply]
Any ideas @dlemstra?
I looked into using Distort SRT but the result looks odd (some weird artifacts) but would be great if this is something I can do at compositing and just tell it to rotate then and there.

**Comment by dlemstra** [comment]
I haven't had time to check this yet.

**Comment by dlemstra** [comment]
There is no build-in way to do this inside ImageMagick you will be forced to do the math yourself:

```C#
var angle = 45;
var radians = Math.PI * 45 / 180.0;
var offset = dropImage.Height * Math.Sin(radians);

dropImage.BackgroundColor = MagickColors.Transparent;
dropImage.Rotate(angle);

var position = new PointD(drop.GetXCoordinate() - offset, drop.GetYCoordinate());
baseImage.Composite(dropImage, position, CompositeOperator.SrcOver);
```

**Reply by CommanderAlchemy** [reply]
Hi thanks for this sorry for the late reply. This works for some angles but not all. 
My trigonometry is a bit lacking here but how should I handle all possible 360 degrees?

**Reply by dlemstra** [reply]
I will leave that up to you. This is something that is not related to this library.
